VSDiagnostics/VSDiagnostics/VSDiagnostics.Test/Tests/General/ImplementEqualsAndGetHashCodeTests.cs

```diff
@@ -1421,5 +1421,136 @@ public override int GetHashCode()
             VerifyDiagnostic(original, "Class MyClass does not implement Equals() and GetHashCode().");
             VerifyFix(original, result);
         }
+
+        [TestMethod]
+        public void ImplementEqualsAndGetHashCode_ClassDoesNotImplementEither_HasExpressionBodiedProperty()
+        {
+            var original = @"
+namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        string Foo => ""test"";
+    }
+}";
+
+            var result = @"
+namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        string Foo => ""test"";
+
+        public override bool Equals(object obj)
+        {
+            if (obj == null || typeof(MyClass) != obj.GetType())
+            {
+                return false;
+            }
+
+            var value = (MyClass)obj;
+            return Foo == value.Foo;
+        }
+
+        public override int GetHashCode()
+        {
+            // Add any fields you're interested in, taking into account the guidelines described in
+            // https://msdn.microsoft.com/en-us/library/system.object.gethashcode%28v=vs.110%29.aspx
+            return base.GetHashCode();
+        }
+    }
+}";
+
+            VerifyDiagnostic(original, "Class MyClass does not implement Equals() and GetHashCode().");
+            VerifyFix(original, result);
+        }
+
+        [TestMethod]
+        public void ImplementEqualsAndGetHashCode_PartialClassDoesNotImplementEither_ImplementsAllTypesInSplitClass()
+        {
+            var original = @"
```

Comment:
These should be put in separate trees instead of one syntax tree with multiple classes. For example: https://github.com/VSDiagnostics/VSDiagnostics/blob/develop/VSDiagnostics/VSDiagnostics/VSDiagnostics.Test/Tests/General/RedudantPrivateSetterTests.cs#L226